cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -482,7 +482,7 @@ WorkflowStatistics getWorkflowStatistics(Id.Workflow workflowId, long startTime,
    * @param timeInterval The timeInterval with which the user wants to space out the runs
    * @return Map of runId of Workflow to DetailedStatistics of the run
    */
-  Collection<WorkflowDataset.WorkflowRunRecord> retrieveSpacedRecords(Id.Workflow workflow, String runId,
+  Collection<WorkflowDataset.WorkflowRunRecord> retrieveSpacedRecords(Id.Program workflow, String runId,
```

Comment:
Should we make it simple and then just remove ``Id.Workflow`` and other classes? It seems weird to me (probably because I've now gotten used to it :-) ), that ``Id.Workflow``, etc does not extend ``Id.Program``. If they are supposed to be helper classes, lets move them outside ``Id`` itself, into a helper ``Programs`` class. But even that is an overkill for saving on just a single ``ProgramType`` parameter.